[lineage] createLineageRuntimeInstance is declared in dist types but missing from JS exports

## Summary

`@manifesto-ai/lineage` 5.0.0 ships `dist/lineage-runtime.d.ts` declaring:

```ts
export declare function createLineageRuntimeInstance<T extends ManifestoDomainShape>(
  kernel: LineageRuntimeKernel<T>,
  service: ResolvedLineageConfig["service"],
  config: ResolvedLineageConfig,
  view?: RuntimeExecutionView<T>,
  isView?: boolean,
): LineageInstance<T>;
```

…but the implementation is bundled into an internal chunk and **not exported from any JS entrypoint**:

```
$ node -e "import('@manifesto-ai/lineage/provider').then(m => console.log(Object.keys(m)))"
[ 'DefaultLineageService', 'InMemoryLineageStore', 'attachLineageDecoration',
  'createInMemoryLineageStore', 'createLineageRuntimeController',
  'createLineageService', 'getLineageDecoration' ]

$ node -e "import('@manifesto-ai/lineage/dist/lineage-runtime.js')"
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '.../dist/lineage-runtime.js'
```

## Why it matters

This is asymmetric with the base SDK, whose equivalent **is** public on the provider seam:

```ts
// @manifesto-ai/sdk/provider
export { createBaseRuntimeInstance } from "./runtime/base-runtime.js";
```

A decorator/provider author who composes a runtime around a pre-built kernel (e.g. to hydrate a snapshot via `setVisibleSnapshot` before wrapping) can do this in base mode but not in lineage mode — the only public composition paths are `withLineage(...).activate()` (which creates its own kernel via the factory) or the lower-level `createLineageRuntimeController` (seal/continuity only, no app surface).

We hit this while evaluating lineage adoption in manifesto-ai/studio. We ultimately did not need it (pre-sealing genesis through `LineageService.prepareSealGenesis` + `restore()` covers the hydration case), but the dangling declaration cost a debugging round-trip: TypeScript resolves the import fine and it fails only at runtime with `ERR_MODULE_NOT_FOUND`.

## Suggested resolution

Either:

1. Export `createLineageRuntimeInstance` from `@manifesto-ai/lineage/provider` (mirroring `createBaseRuntimeInstance` on the SDK provider seam), or
2. Remove `lineage-runtime.d.ts` from the published `dist` so the type surface matches the JS surface.

## Environment

- `@manifesto-ai/lineage` 5.0.0, Node v25.9.0

--- a/docs/api/public-surface.md
+++ b/docs/api/public-surface.md
@@ -1018,6 +1018,7 @@ _Source: `packages/lineage/src/provider.ts`_
 - `attachLineageDecoration`
 - `createInMemoryLineageStore`
 - `createLineageRuntimeController`
+- `createLineageRuntimeInstance`
 - `createLineageService`
 - `DefaultLineageService`
 - `getLineageDecoration`

--- a/packages/lineage/src/__tests__/provider-exports.test.ts
+++ b/packages/lineage/src/__tests__/provider-exports.test.ts
@@ -1,0 +1,28 @@
+import { describe, expect, it } from "vitest";
+
+import * as provider from "../provider.js";
+
+/**
+ * Regression test for #491: the dist types declared
+ * createLineageRuntimeInstance but no JS entrypoint exported it, so
+ * TypeScript resolved the import and Node failed at runtime with
+ * ERR_MODULE_NOT_FOUND. The provider seam must export it (mirroring
+ * createBaseRuntimeInstance on the SDK provider seam).
+ */
+describe("@manifesto-ai/lineage/provider exports (#491)", () => {
+  it("exports createLineageRuntimeInstance as a function", () => {
+    expect(typeof provider.createLineageRuntimeInstance).toBe("function");
+  });
+
+  it("keeps the established provider seam surface", () => {
+    for (const name of [
+      "attachLineageDecoration",
+      "createLineageRuntimeController",
+      "getLineageDecoration",
+      "createLineageService",
+      "createInMemoryLineageStore",
+    ]) {
+      expect(typeof provider[name as keyof typeof provider]).toBe("function");
+    }
+  });
+});

--- a/packages/lineage/src/provider.ts
+++ b/packages/lineage/src/provider.ts
@@ -41,4 +41,8 @@ export {
   DefaultLineageService,
   createLineageService,
 } from "./service/lineage-service.js";
+// Mirrors createBaseRuntimeInstance on the SDK provider seam: decorator and
+// provider authors can compose a lineage runtime around a pre-built kernel
+// (#491 — the declaration shipped in dist types without a JS export).
+export { createLineageRuntimeInstance } from "./lineage-runtime.js";
 export { InMemoryLineageStore, createInMemoryLineageStore } from "./store/in-memory-lineage-store.js";

